addons/l10n_fr_pos_cert/static/src/js/l10n_fr_pos_cert.js

```diff
@@ -0,0 +1,244 @@
+    var core = require('web.core');
+    var _t = core._t;
+
+    /*************************************************************************
+        Promise that will be resolved, when the hash of the saved order is
+        known
+     */
+    var certification_deferred = null;
+
+    /*************************************************************************
+     * Function that return certification text, depending of a hash
+     * @param {string} hash: value of the pos_order.l10n_fr_hash
+     * @param {string} setting: value of the pos_config.l10n_fr_print_hash
+     * @returns {string}: Certification Text that will be printed on the bill
+     */
+    var prepare_certification_text = function(hash, setting){
+        if (setting === 'no_print'){
+            return '';
+        }
+        if (hash){
+            return _t('Certification Number: ') + hash.substring(0, 10) + '...' + hash.substring(hash.length - 10);
+        }
+        return _t("Because of a network problem, this ticket could not be certified.");
+    };
+
+    /*************************************************************************
+        Extend module.Order:
+            Add a new certification_text field that will content a text and
+            the hash of the PoS Order, or a warning if hash has not been
+            recovered.
+     */
+    var OrderParent = models.Order.prototype;
+    models.Order = models.Order.extend({
+        set_hash: function(hash, setting) {
+            var certification_text = prepare_certification_text(hash, setting);
+            this.set({
+                hash: hash,
+                certification_text: certification_text,
+            });
+        },
+
+        export_for_printing: function(attributes){
+            var order = OrderParent.export_for_printing.apply(this, arguments);
+            if (this.pos.config.l10n_fr_print_hash === 'no_print'){
+                order.certification_text = '';
+            } else {
+                // We add a tag that will be replaced after, because
+                // when export_for_printing is called, hash is unknown
+                order.certification_text = "__CERTIFICATION_TEXT__";
+            }
+            return order;
+        },
+
+        export_as_JSON: function() {
+            var order = OrderParent.export_as_JSON.apply(this, arguments);
+            order.certification_text = this.get('certification_text');
+            return order;
+        },
+
+    });
+
+    /*************************************************************************
+        Extend module.PosModel:
+            - Overload load_server_data(). Load extra certification settings
+              at the end of the PoS load.
+            - Overload _save_to_server() and store if the order has been
+              correctly created in the promise 'certification_deferred'
+     */
+    var PosModelParent = models.PosModel.prototype;
+    models.PosModel = models.PosModel.extend({
```

Comment:
why not posModel.include ?

Reply:
@legalsylvain 
Woops, my bad.
PosModels is a Backbone class, and I guess the way to extend their behavior is indeed what you do.

On the other hand, PosWidgets like ProxyDevice are regular OdooClass extension, which should support
`this._super.apply(this, arguments)` instead of `prototype.function()`